### ISSUE TITLE
Electron 0.36.0 fullscreen shows titlebar on OS X Yosemite

### ISSUE THREAD

**Issue by iamdriz** [atom/electron#3811]
The latest version of Electron (0.36.0) still has an issue on Yosemite with fullscreen.

Basically when in fullscreen mode the window shows a textured title at the top:

![screen shot 2015-12-14 at 16 53 11](https://cloud.githubusercontent.com/assets/801252/11787461/b537ec3c-a283-11e5-9591-dae3ee0672bf.png)

This happens regardless if the window frame option is set to true or false (the only difference being you don't see the traffic lights on hover when set to false). However setting the transparent option to true removes the titlebar completely and solves it.

So in short it needs to hide the titlebar when transparency is set to false and in fullscreen and then ONLY show the title when hovering and the frame is set to true.

**Comment by zcbenz** [comment]
Is this the same issue with #1057? I think the problem is no one else can reproduce this.

Also can you try setting the `type` attribute of `BrowserWindow` to `textured`?

**Comment by asherkin** [comment]
I get similar on Linux if `resizable` is set to `false` (as it is in #1057 – which makes sense to me).

Wonder if it's the same root issue...  
(and my apologies for the noise if it isn't :P)

**Comment by iamdriz** [comment]
@zcbenz Tried it with type set to textured, but still get the issue. This is the code to reproduce:

~~~
app.on('ready', function() {
  mainWindow = new BrowserWindow({
    'width': 800,
    'height': 600,
    'autoHideMenuBar': true,
    'useContentSize': true,
    'titleBarStyle': 'hidden-inset',
    'frame': false,
    'fullscreen': true,
    'type': 'textured'
  });
  mainWindow.loadURL('file://' + __dirname + '/index.html');
  mainWindow.focus();
});
~~~

**Comment by iamdriz** [comment]
Okay just found out some more information!

Basically if you set the titleBarStyle (even though the frame is set to false) it will show the titlebar all the time like the screenshots above. This is a bug as I think once frame is set to false, anything related to the frame such as traffic light buttons should be ignored.

However there is still a slight issue (even when removing the titleBarStyle) as when you hover to the top to access the menu bar you see this:

This happens when frame is set to true and false. It should only appear if the frame is set to true when hovering. Setting the frame to be transparent however solves this as well, but these are all workarounds and not expected behaviour IMO.

**Comment by zcbenz** [comment]
Didn't know you were using the `titleBarStyle` option, this is actually a duplicate of #2810.
